packages/typespec-client-generator-core/design-docs/client.md

```diff
@@ -0,0 +1,379 @@
+It is better to consolidate `@client` and `@operationGroup`.
+
+## New proposal
+
+1. Change `@clientInitialization` decorator and add `clientInitialization` property to `SdkClientType`
+- Change `@clientInitialization` decorator's `options` parameter to `ClientInitializationOptions` type to accept `access` and `accessorAccess` setting.
+- Add `clientInitialization` property to `SdkClientType`. It indicate the initialization method's signature, including the parameter list and access.
+- When `access` set to `Access.public`, it means this client could be initialized separately, while `Access.internal` could not. Default to `Access.public` for client and `Access.internal` for sub client.
+- When `accessorAccess` set to `Access.public`, it means this client could be get from parent, while `Access.internal` could not. Default to `undefined` for client and `Access.public` for sub client.
+
+For example:
+```typespec
+@service({
+  title: "Pet Store",
+})
+namespace PetStore {
+  @route("/dogs")
+  interface Dogs {
+    feed(): void;
+    pet(): void;
+  }
+
+  @route("/cats")
+  namespace Cats {
+    op feed(): void;
+    op pet(): void;
+  }
+}
+
+@client({
+  name: "DogsClient",
+  service: PetStore,
+})
+namespace DogsClient {
+  @client
+  interface Feed {
+    feed is PetStore.Dogs.feed;
+  }
+
+  @client
+  interface Pet {
+    pet is PetStore.Dogs.pet;
+  }
+}
+
+@client({
+  name: "CatsClient",
+  service: PetStore,
+})
+namespace CatsClient {
+  @client
+  interface Feed {
+    feed is PetStore.Cats.feed;
+  }
+
+  @client
+  interface Pet {
+    pet is PetStore.Cats.pet;
+  }
+}
+```
+The above tsp gets the two root clients: `DogsClient` and `CatsClient`. 
+All of them has two client accessor method to get sub clients, and with `public` access.
+All of them has two sub clients: `Feed` and `Pet`.
+The initialization method of `Feed` and `Pet` is `internal`.
+
+Another example with customization:
+```typespec
+@service({
+  title: "My Service",
+})
+namespace MyService {
+  interface InnerGroup {
+    op upload(@path blobName: string): void;
+  }
+}
+
+namespace MyCustomizations {
+  model InnerGroupClientOptions {
+    blobName: string;
+  }
+
+  @@clientInitialization(InnerGroup, {parameters: InnerGroupClientOptions, access: Access.public, accessorAccess: Access.internal});
+}
+```
+The above tsp gets client `MyServiceClient` and sub client `InnerGroup`. 
+The `MyServiceClient` has a client accessor method whose response is `InnerGroup`. The method has a parameter named `blobName`. The method is `internal`.
+The `InnerGroup`'s `initialization` model's properties contains a property named `blobName`. The method is `public`.
+The method `upload` no longer has `blobName` parameter, its corresponding operation's parameter `blobName` is mapped to the `blobName` property of `InnerGroup`'s `initialization` model.
+
+2. Move client accessor method out of the normal methods list of client and add a new `clientAccessor` property to put it.
```

Comment:
will the `clientAccessor` be on the parent client or the child client? Off the top of my head, we want to know a client's subclients, so I think it would be better to have this on the parent

Reply:
it is on the parent's `getXXX` method.